test/script/ReallocateScUsdcV2.s.t.sol

```diff
@@ -0,0 +1,284 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.19;
+
+import "forge-std/console2.sol";
+import "forge-std/Test.sol";
+import {FixedPointMathLib} from "solmate/utils/FixedPointMathLib.sol";
+
+import {scUSDCv2} from "../../src/steth/scUSDCv2.sol";
+import {PriceConverter} from "../../src/steth/PriceConverter.sol";
+import {IAdapter} from "../../src/steth/IAdapter.sol";
+import {AaveV2ScUsdcAdapter} from "../../src/steth/scUsdcV2-adapters/AaveV2ScUsdcAdapter.sol";
+import {AaveV3ScUsdcAdapter} from "../../src/steth/scUsdcV2-adapters/AaveV3ScUsdcAdapter.sol";
+import {MorphoAaveV3ScUsdcAdapter} from "../../src/steth/scUsdcV2-adapters/MorphoAaveV3ScUsdcAdapter.sol";
+import {ReallocateScUsdcV2} from "../../script/v2/keeper-actions/ReallocateScUsdcV2.s.sol";
+import {MainnetAddresses} from "../../script/base/MainnetAddresses.sol";
+import {Constants} from "../../src/lib/Constants.sol";
+
+contract ReallocateScUsdcV2Test is Test {
+    using FixedPointMathLib for uint256;
+
+    uint256 mainnetFork;
+
+    scUSDCv2 vault;
+    AaveV3ScUsdcAdapter aaveV3;
+    AaveV2ScUsdcAdapter aaveV2;
+    MorphoAaveV3ScUsdcAdapter morpho;
+    PriceConverter priceConverter;
+    ReallocateScUsdcV2TestHarness script;
+
+    function setUp() public {
+        mainnetFork = vm.createFork(vm.envString("RPC_URL_MAINNET"));
+        vm.selectFork(mainnetFork);
+        vm.rollFork(17987643);
+
+        script = new ReallocateScUsdcV2TestHarness();
+
+        vault = scUSDCv2(MainnetAddresses.SCUSDCV2);
+        priceConverter = vault.priceConverter();
+        morpho = script.morphoAdapter();
+        aaveV2 = script.aaveV2Adapter();
+        aaveV3 = script.aaveV3Adapter();
+
+        _initialRebalance();
+    }
+
+    function test_run_moveHalfOfThePositionFromMorphoToAaveV2() public {
+        assertTrue(vault.totalDebt() > 0, "vault has no debt");
+        assertTrue(vault.totalCollateral() > 0, "vault has no collateral");
+
+        assertEq(_getAllocationPercent(morpho), 0.5e18, "morpho allocation percent");
+        assertEq(_getAllocationPercent(aaveV2), 0.5e18, "aave v2 allocation percent");
+
+        uint256 morphoInitialCollateral = vault.getCollateral(morpho.id());
+        uint256 morphoInitialDebt = vault.getDebt(morpho.id());
+        uint256 aaveV2InitialCollateral = vault.getCollateral(aaveV2.id());
+        uint256 aaveV2InitialDebt = vault.getDebt(aaveV2.id());
+
+        assertTrue(script.useMorpho(), "morpho not used");
+        assertTrue(script.useAaveV2(), "aave v2 not used");
+
+        uint256 morphoAllocationPercent = 0.25e18;
+        uint256 aaveV2AllocationPercent = 0.75e18;
+
+        uint256 expectedMorphoWithdrawAmount = morphoInitialCollateral / 2;
+        uint256 expectedMorphoRepayAmount = morphoInitialDebt / 2;
+
+        script.setMorphoAllocationPercent(morphoAllocationPercent);
+        script.setAaveV2AllocationPercent(aaveV2AllocationPercent);
+
+        script.run();
+
```

Comment:
Can you also add an assertion that the totalAssets does not change after reallocation?

(for all the tests)

Reply:
[added](https://github.com/lindy-labs/sandclock-contracts/pull/123/commits/d4db4d4252d4298d7147682c89b5d963c4664239)